server/src/cron/caching/getOneMenu.js

```diff
@@ -0,0 +1,47 @@
+import axios from "axios";
+
+import getMenuItemList from "./getMenuItemList";
+
+import mealNames from "../../config/mealNames";
+import locations from "../../config/locations";
+import dateBuilder from "../../util/dateBuilder";
+import queryBuilder from "../../util/queryBuilder";
+import { MENUS_URI, YD_VERSION, E_NO_API_RES } from "../../config/constants";
+
+export default async function getOneMenu(location) {
+    const endpoint =
+        MENUS_URI + queryBuilder({ version: YD_VERSION, location });
+    const response = await axios.get(endpoint);
+    const data = response.data;
+    // throw on bad response from Yale Dining
+    if (!data || !data.DATA || !data.DATA.length || !data.DATA[0].length) {
+        console.error(`${E_NO_API_RES}: ${locations[location]}`);
+        return undefined;
+    }
+    console.log(`Populating ${locations[location]}...`);
+    const today = dateBuilder(0);
+    const tomorrow = dateBuilder(1);
+    var menus = {};
+    menus["location"] = locations[location];
+    menus["today"] = {};
+    menus["tomorrow"] = {};
+    for (let mealName in mealNames) {
+        const filteredData = data.DATA.filter(
+            entry => entry[data.COLUMNS.indexOf("MEALNAME")] === mealName
+        );
+        menus["today"][mealNames[mealName]] = await getMenuItemList(
+            data.COLUMNS,
+            filteredData.filter(
+                entry => entry[data.COLUMNS.indexOf("MENUDATE")] === today
+            )
+        );
+        menus["tomorrow"][mealNames[mealName]] = await getMenuItemList(
+            data.COLUMNS,
+            filteredData.filter(
+                entry => entry[data.COLUMNS.indexOf("MENUDATE")] === tomorrow
+            )
+        );
+    }
+    console.log(`Populated ${locations[location]}.`);
+    return menus;
+}
```

Comment:
So will this (along with the other files in cron/caching) totally replace router/MenusRouter and router/MenuItemsRouter?

Reply:
Not quite. Sanat's commit will do that. This just populates our Firestore database with the new schema, which will be used to replace router/MenusRouter. router/MenuItemsRouter will be deprecated.